runtime/lua/vim/filetype/detect.lua

```diff
@@ -270,6 +308,38 @@ function M.dep3patch(path, bufnr)
   end
 end
 
+function M.diff(contents)
```

Comment:
Passing `contents` directly to these functions is a good step toward allowing both `buf` and `contents` keys to `vim.filetype.match` (as discussed [here](https://github.com/neovim/neovim/pull/18710#issuecomment-1167318027)). 👍